go/cmd/server/server.go

```diff
@@ -49,21 +52,87 @@ func (s *servingServiceServer) GetOnlineFeatures(ctx context.Context, request *s
 			FeatureNames: &serving.FeatureList{Val: make([]string, 0)},
 		},
 	}
+	// Entities are currently part of the features as a value and the order that we add it to the resp MetaData
+	// Need to figure out a way to map the correct entities to the correct ordering
+	entityValuesMap := make(map[string][]*prototypes.Value, 0)
+	featureNames := make([]string, len(featureVectors))
+	for idx, vector := range featureVectors {
 
-	for _, vector := range featureVectors {
 		resp.Metadata.FeatureNames.Val = append(resp.Metadata.FeatureNames.Val, vector.Name)
-
+		featureNames[idx] = vector.Name
 		values, err := types.ArrowValuesToProtoValues(vector.Values)
 		if err != nil {
 			return nil, err
 		}
-
+		if _, ok := request.Entities[vector.Name]; ok {
+			entityValuesMap[vector.Name] = values
+		}
 		resp.Results = append(resp.Results, &serving.GetOnlineFeaturesResponse_FeatureVector{
 			Values:          values,
 			Statuses:        vector.Statuses,
 			EventTimestamps: vector.Timestamps,
 		})
 	}
 
+	go generateLogs(s, entityValuesMap, featureNames, resp.Results, request.RequestContext)
 	return resp, nil
 }
+
+func generateLogs(s *servingServiceServer, entityMap map[string][]*prototypes.Value, featureNames []string, features []*serving.GetOnlineFeaturesResponse_FeatureVector, requestData map[string]*prototypes.RepeatedValue) error {
+	// Add a log with the request context
+	if len(requestData) > 0 {
+		requestContextLog := logging.Log{
+			RequestContext: requestData,
+		}
+		s.loggingService.EmitLog(&requestContextLog)
+	}
+
+	if len(features) <= 0 {
+		return nil
+	}
+
+	entityArr, err := s.fs.ListEntities(true)
+	if err != nil {
+		return err
+	}
+
+	joinKeys := make([]string, 0)
+	for _, entity := range entityArr {
+		joinKeys = append(joinKeys, entity.JoinKey)
+	}
+
+	numFeatures := len(featureNames)
+	// Should be equivalent to how many entities there are(each feature row has (entity) number of features)
+	// acc_rate []
+	numRows := len(features[0].Values)
+	featureValueLogRows := make([][]*prototypes.Value, numRows)
```

Comment:
I don't get it, why we build these arrays if they are not used? In each loop we emit one row and then newer use it again.

Reply:
Oh this was left over and I didnt' realize I needed to change the implementation. Fixed.